src/netlink_namespace.c

```diff
@@ -688,18 +693,24 @@ static int ir_config(const char *key, const char *value) {
   return (status);
 } /* int ir_config */
 
-static void close_fds(int* selfns, int* targetns) {
-  if ((*selfns >= 0) && (close(*selfns) != 0)) {
-    ERROR("Closing \"selfns\" file failed: %m");
-  }
+static void ir_init_error(int* selfns, int* targetns) {
   if ((*targetns >= 0) && (close(*targetns) != 0)) {
     ERROR("Closing \"targetns\" file failed: %m");
   }
-} /* fd_close */
+  if (*selfns >= 0) {
+    if (setns(*selfns, CLONE_NEWNET) != 0) {
+      ERROR("netlink_namespace plugin: ir_init_error: Getting back to original namespace failed: %m");
```

Comment:
Similarly here - you're now in a slightly tricky place, you're trying to revert back to the original network namespace. Leaving it behind would be actively harmful. If this fails, it's abort() time. However - it might be worth checking whether or not setting the network namespace to the current network namespace - does it return an error or does it succeed ? (Either by checking docs or experiment).